modules/javafx.controls/src/main/java/javafx/scene/control/skin/PaginationSkin.java

```diff
@@ -649,7 +651,11 @@ private void resetIndexes(boolean usePageIndex) {
         currentStackPane.getChildren().clear();
         nextStackPane.getChildren().clear();
 
-        getSkinnable().setCurrentPageIndex(currentIndex);
+        if (usePageIndex) {
```

Comment:
Shouldn't this be `!usePageIndex`, given the stated intent? The `resetIndexes` method is called twice: once from the constructor with `usePageIndex = true` and once from `resetIndiciesAndNav` with `usePageIndex = false`.